schnell/space_detector.py

```diff
@@ -0,0 +1,369 @@
+        return tr
+
+class LISAandALIADetector(Detector):
+    """ :class:`LISAandALIADetector objects can be used to describe 
+    the combination of a LISA and an ALIA network. id 0 to 2 is LISA;
+    id 3 to 5 is ALIA.
+    """
+    trans_freq_earth = 2 * np.pi / (365 * 24 * 3600)
+    R_AU = 1.496E11
+    kap = 0  # initial longitude of LISA ; ALIA is behind
+    lam = 0  # initial orientation
+    clight = 299792458.
+
+    def __init__(self, detector_id, is_L5Gm=False,
+                 static=False, include_GCN=False,
+                 mission_duration=4., separation=0.7):
+        self.separation = separation # distance between LISA and ALIA, in AU
+        # We suppose that the Earth's motion is circular (?)
+        self.ang_separation = 2 * np.arcsin(self.separation / 2)
+
+        if detector_id in [0, 1, 2]:
+            self.detector = LISADetector2(detector_id, is_L5Gm, static,
+                                          include_GCN, mission_duration)
+            self.name = 'LISA_%d' % detector_id
+            self.get_transfer = self.detector._get_transfer_LISA
+            self.detector.kap = self.kap
+        elif detector_id in [3, 4, 5]:
+            self.detector = ALIADetector2(detector_id-3, static, include_GCN, mission_duration)
+            self.name = 'ALIA_%d' % detector_id
+            self.get_transfer = self.detector._get_transfer_ALIA
+            self.detector.kap = self.kap - self.ang_separation
+        else:
+            raise RuntimeError('Detector i_d must be between 0 and 5')
+        self.is_L5Gm = is_L5Gm
+        self.detector.lam = self.lam
+        self.i_d = detector_id
+        self.L = self.detector.L
+        self.e = self.detector.e
+    
+    def psd_A(self, f):
+        return self.detector.psd_A(f)
+    
+    def psd_X(self, f):
+        return self.detector.psd_X(f)
+    
+    def GCN(self, f):
+        return self.detector.GCN(f)
+    
+    def psd(self, f):
+        return self.detector.psd(f)
+    
+    def response(self, f):
+        return self.detector.response(f)
+    
+    def sensitivity(self, f):
+        return self.detector.sensitivity(f)
+    
+    def get_position(self, t):
+        """ Returns a 2D array containing the 3D position of
+        the detector at a series of times. The output array
+        has shape [3, N_t], where N_t is the size of `t`.
+
+        .. note:: The spacecraft orbits are calculated using Eq. 1
+                  of gr-qc/0311069.
+
+        Args:
+            t: time of observation (in seconds).
+
+        Returns:
+            array_like: detector position (in m) as a function \
+                of time.
+        """
+        return self._pos_single(t, self.i_d)
+    
+    def _pos_all(self, t):
+        return np.array([self._pos_single(t, i)
+                         for i in range(6)])
+    
+    def _pos_single(self, t, n):
+        if n in [0, 1, 2]:
+            detect = LISADetector2(0, is_L5Gm=self.is_L5Gm,
```

Comment:
Humm, could you have used `self.detector` instead if you checked `if n == self.i_d`? Just wondering if having to define a new detector every time will slow things down.

Reply:
I am not sure how much time we would gain, but it's really easy to do so I will probably implement this soon